### ISSUE TITLE
Use aapt from Android studio is not found in PATH

### ISSUE THREAD

**Issue by trufae** [sensepost/objection#244]
Default installation of Android Studio (on Mac) is this:

```
$ pwd
~/Library/Android/sdk/build-tools/28.0.3
$ file aapt
aapt: Mach-O 64-bit executable x86_64
```

It will be good if objection can just pick the bin from there, that will make initial setups easier for most users

**Comment by amagdaraog** [comment]
Btw, how do we change the destination for objection to see the aapt location?

**Comment by leonjza** [comment]
It checks `$PATH` for `aapt`, so just modify `$PATH` to include it's location.

**Comment by trufae** [comment]
Yep that works, but i  was suggesting to make objection find those tools in default installation places, feel free to close the issue if you prefer to keep it like this

**Comment by leonjza** [comment]
Will keep this open. Makes sense to try some sane defaults, but `$PATH` modifications will always be the last resort.

**Comment by amagdaraog** [comment]
Can you walk me through how to find $PATH and how to modify to include the directory? Cheers

**Comment by leonjza** [comment]
There are plenty of resources available online for this. But in short, you would do something like `export PATH=$PATH:/path/to/aapt`.

**Comment by amagdaraog** [comment]
Many thanks! But it seems to have an error patching the APK, tried with --skip-resources then it went smooth. But when I open the app, it doesn't seem to hook with the objection.

**Comment by amagdaraog** [comment]
Error without --skip-resources was: 

`No architecture specified. Determining it using `adb`...
Detected target device architecture as: arm64-v8a
Using latest Github gadget version: 12.6.10
Patcher will be using Gadget version: 12.6.10
Unpacking app-ph-debug.apk
An error may have occurred while extracting the APK.
S: WARNING: Could not write to (/Users/adrianemagdaraog/Library/apktool/framework), using /var/folders/hd/pcgdyv2j3mj18brxj2ymhcwc0000gp/T/ instead...
S: Please be aware this is a volatile directory and frameworks could go missing, please utilize --frame-path if the default storage directory is unavailable

App already has android.permission.INTERNET
Target class not specified, searching for launchable activity instead...
Smali not found in smali directory. This might be a multidex APK. Searching...
Found smali at: /var/folders/hd/pcgdyv2j3mj18brxj2ymhcwc0000gp/T/tmpcy19pe25.apktemp/smali_classes22/com/squareup/leakcanary/internal/DisplayLeakActivity.smali
Reading smali from: /var/folders/hd/pcgdyv2j3mj18brxj2ymhcwc0000gp/T/tmpcy19pe25.apktemp/smali_classes22/com/squareup/leakcanary/internal/DisplayLeakActivity.smali
Injecting into an existing constructor
Injecting loadLibrary call at line: 49
Attempting to fix the constructors .locals count
Current locals value is 1, updating to 2:
Writing patched smali back to: /var/folders/hd/pcgdyv2j3mj18brxj2ymhcwc0000gp/T/tmpcy19pe25.apktemp/smali_classes22/com/squareup/leakcanary/internal/DisplayLeakActivity.smali
Copying Frida gadget to libs path...
Patching paused. The next step is to rebuild the APK. If you require any manual fixes, the current temp directory is:
/var/folders/hd/pcgdyv2j3mj18brxj2ymhcwc0000gp/T/tmpcy19pe25.apktemp`

apktool = 2.4.0

**Comment by leonjza** [comment]
Unfortunately I can't see the problem in that output. Ultimately, you will need to manually debug why apktool fails. Also, please don't add more comments to this feature request. It is unrelated.
